pkg/controller/nodes/task/handler.go

```diff
@@ -579,10 +594,23 @@ func (t Handler) Handle(ctx context.Context, nCtx handler.NodeExecutionContext)
 	// STEP 4: Send buffered events!
 	logger.Debugf(ctx, "Sending buffered Task events.")
 	for _, ev := range tCtx.ber.GetAll(ctx) {
```

Comment:
not sure if I need this additional check for buffered events because presumably they wouldn't include a prior success (terminal) event - but added to be safe